for-developers/walkthrough/dev-setup/making-your-first-changes.md

```diff
@@ -47,4 +47,1930 @@ Now you should see some custom content in the post that you added to the feed. Y
+	if err = amountEqualsAdditionalOutputs(spendAmount, additionalOutputs); err != nil {
+		return nil, 0, 0, 0, fmt.Errorf("CreateReactTxn: %v", err)
+	}
+
+	return txn, totalInput, changeAmount, fees, nil
+}
+```
+
+### Creating custom reaction files 
+#### `block_view_reaction.go`
+`block_view_reaction.go` is inspired from `block_view_like.go`.
+It defines setters and getters to the state generated by a `React` transaction to the UTXO view,
+and implements custom `_connectReact` and `disconnectReact` transactions.
+
+```go
+package lib
+
+import (
+	"fmt"
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+	"reflect"
+)
+
+func (bav *UtxoView) _getReactionEntryForReactionKey(reactionKey *ReactionKey) *ReactionEntry {
+	// If an entry exists in the in-memory map, return the value of that mapping.
+	mapValue, existsMapValue := bav.ReactionKeyToReactionEntry[*reactionKey]
+	if existsMapValue {
+		return mapValue
+	}
+
+	// If we get here it means no value exists in our in-memory map. In this case,
+	// defer to the db. If a mapping exists in the db, return it. If not, return
+	// nil. Either way, save the value to the in-memory view mapping got later.
+	reactionExists := false
+	if bav.Postgres != nil {
+		reactionExists = bav.Postgres.GetReaction(reactionKey.ReactorPubKey[:], &reactionKey.ReactedPostHash) != nil
+	} else {
+		reactionExists = DbGetReactorPubKeyToPostHashMapping(bav.Handle, reactionKey.ReactorPubKey[:], reactionKey.ReactedPostHash) != nil
+	}
+
+	if reactionExists {
+		reactionEntry := ReactionEntry{
+			ReactorPubKey:   reactionKey.ReactorPubKey[:],
+			ReactedPostHash: &reactionKey.ReactedPostHash,
+		}
+		bav._setReactionEntryMappings(&reactionEntry)
+		return &reactionEntry
+	}
+
+	return nil
+}
+
+func (bav *UtxoView) _setReactionEntryMappings(reactionEntry *ReactionEntry) {
+	// This function shouldn't be called with nil.
+	if reactionEntry == nil {
+		glog.Errorf("_setReactionEntryMappings: Called with nil ReactionEntry; " +
+			"this should never happen.")
+		return
+	}
+
+	reactionKey := MakeReactionKey(reactionEntry.ReactorPubKey, *reactionEntry.ReactedPostHash)
+	bav.ReactionKeyToReactionEntry[reactionKey] = reactionEntry
+}
+
+func (bav *UtxoView) _deleteReactionEntryMappings(reactionEntry *ReactionEntry) {
+
+	// Create a tombstone entry.
+	tombstoneReactionEntry := *reactionEntry
+	tombstoneReactionEntry.isDeleted = true
+
+	// Set the mappings to point to the tombstone entry.
+	bav._setReactionEntryMappings(&tombstoneReactionEntry)
+}
+
+func (bav *UtxoView) GetReactionByReader(readerPK []byte, postHash *BlockHash) bool {
+	// Get react state.
+	reactionKey := MakeReactionKey(readerPK, *postHash)
+	reactionEntry := bav._getReactionEntryForReactionKey(&reactionKey)
+	return reactionEntry != nil && !reactionEntry.isDeleted
+}
+
+func (bav *UtxoView) GetReactionsForPostHash(postHash *BlockHash) (_ReactorPubKeys [][]byte, _err error) {
+	if bav.Postgres != nil {
```

Comment:
we should refactor this function to use the `DbAdapter` to get the reactions from the db and only update the view if the key constructed from the entry does not exist in the view yet. Otherwise, we risk updating entries in the view that haven't been flushed.